### PR TITLE
refactor 2d inner reduction heuristics

### PR DESCRIPTION
**Pull request by liqiangxl**
Following https://github.com/NVIDIA/Fuser/pull/3331/
refactor 2d inner reduction heuristics, prepare for further heuristics tunning.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!test --diff --diff-bench

**Comment by jjsjann123** [comment]
> Following #3331 refactor 2d inner reduction heuristics, prepare for further heuristics tunning.

should the target of this one be `llu/clean_2d_inner_reduction_heuristics` instead? That'll make review easier.

**Comment by liqiangxl** [comment]
!test --diff --diff-bench

**Comment by liqiangxl** [comment]
> > Following #3331 refactor 2d inner reduction heuristics, prepare for further heuristics tunning.
> 
> should the target of this one be `llu/clean_2d_inner_reduction_heuristics` instead? That'll make review easier.

Yes, `clean_2d_inner_reduction_heuristics` is merged into main and this PR is then rebased on main. It is ready for review.

**Review by jjsjann123 — APPROVED** [review]
LGTM, one small question on code change.

File: csrc/scheduler/reduction.cpp [hunk 1]
```diff
@@ -194,129 +200,120 @@ std::unique_ptr<ReductionParams> inner2dReductionHeuristic(
...
+
+  // Parallelization strategy:
+  // [] indicates optional
+  // Reduction dim: Serial, [grdim], TIDx, Vect
+  // Iteration dim: Serial, godim, [TIDy]
+
+  // Max vectorization factor
+  int64_t vect_factor = std::min(
+      scheduler_utils::lastPow2(max_unroll), (int64_t)max_vectorize_factor);
+  int64_t after_vect = total_reduction_numel / vect_factor;
```
[the remark this answers is not in the document]
Comment (liqiangxl) [reply csrc/scheduler/reduction.cpp:240]: Yes, `vect_factor` was derived from inner most dim, `total_reduction_numel` must be disivible by `vect_factor`

**Pull request merged**